### PR TITLE
Mobile: email thread preview on approval for email reply actions (#975 phase 3)

### PR DESCRIPTION
**Pull request by chiedocursor**
## Summary

- Adds `EmailThreadCard` on the approval detail screen for `google.send_email_reply` and `microsoft.send_email_reply`, rendering `context.details.email_thread` (subject, latest message expanded, collapsible earlier messages, attachment filename chips, plain `body_text` with snippet fallback).
- Omits `email_thread` from the generic "Additional Context" key-value list so the thread is not shown twice as raw JSON.
- Adds Jest coverage for the card, parsing helpers, and approval detail wiring.

Refs #975 (Phase 3 only; Phases 1–2 and verification remain elsewhere).

## Related Issue

Refs #975

### Developer

- [x] Unit tests added / updated
- [ ] `make test` passes locally (not run in this environment; mobile-only change)
- [ ] `make build` succeeds
- [x] Mobile: `cd mobile && npm test -- --ci` passes

### Manual Testing

- [ ] Open a pending approval for an email reply action and confirm subject, messages, and attachments display; expand "Earlier in this thread" for multi-message threads; tap "Show more" on truncated messages.

## Screenshots / Recordings

N/A (not captured in agent environment)

## Notes for Reviewers

- `mobile/src/api/schema.d.ts` is gitignored and generated by `npm run generate:api` in `mobile/`; types use `components["schemas"]["EmailThread"]` from that file.

<div><a href="https://cursor.com/agents/bc-6662e79e-5669-4b1d-a768-6a4ad48cee95"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-6662e79e-5669-4b1d-a768-6a4ad48cee95"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by chiedocursor** [comment]
## 🤖 Watch Session Summary

### Changes Made
No changes made during this session.

### Decision Log

#### ✅ Implemented
No review comments acted on.

---
*Watch session ended after 0 minutes of inactivity. Processed 0 comments across 1 poll cycles.*

File: mobile/src/screens/approvals/EmailThreadCard.tsx [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+function AttachmentChips({
+  attachments,
+}: {
+  attachments: NonNullable<EmailThreadMessage["attachments"]>;
+}) {
+  return (
+    <View style={styles.attachRow} accessibilityRole="list">
+      {attachments.map((a) => (
+        <View
+          key={`${a.filename}-${a.size_bytes}`}
```
Comment (chiedoclaude) [inline mobile/src/screens/approvals/EmailThreadCard.tsx:74]: **[Medium · Code Quality]** `key` can collide when two attachments share the same filename and size (e.g. the same file attached twice).

**Why:** Duplicate React keys cause stale/incorrect reconciliation — the wrong attachment chip can persist during re-renders, and React will emit a dev-mode warning.

**Fix:** Include the array index as a tiebreaker:
```tsx
attachments.map((a, i) => (
  <View
    key={`${i}-${a.filename}`}
    ...
  >
```
Index-as-fallback is safe here because the list is server-derived and never reordered in place.

**Review by chiedoclaude — COMMENTED** [review]
## Review Round 1/6

**Findings:** 4 (0 critical, 0 high, 1 medium, 3 low)

The implementation is clean and well-structured. The main concern is a duplicate React key risk in `AttachmentChips`; the rest are low-priority observations.

### What Looks Good
- `parseEmailThreadDetails` is robustly defensive — it handles partial, malformed, or missing API payloads gracefully without throwing.
- Filtering `email_thread` from the generic "Additional Context" list is correctly wired through the `isEmailReply` guard and properly placed in the useMemo dep array.
- Test coverage is strong: 3 new test files covering the utility, card, and screen integration levels, with good edge-case coverage (truncation, collapse/expand, attachments).
- Using OpenAPI-generated types throughout is the right call; no hand-written type aliases.
- Accessibility attributes (`accessibilityRole`, `accessibilityState`, `accessibilityLabel`) are present on all interactive and list elements.

---
*Automated review by `/review` — Round 1 of 6*

File: mobile/src/screens/approvals/EmailThreadCard.tsx [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+function formatSizeBytes(n: number): string {
+  if (n < 1024) return `${n} B`;
+  if (n < 1024 * 1024) return `${(n / 1024).toFixed(1)} KB`;
+  return `${(n / (1024 * 1024)).toFixed(1)} MB`;
+}
+
+function messageBodyText(m: EmailThreadMessage): string {
+  const trimmed = m.body_text?.trim() ?? "";
+  if (trimmed.length > 0) return m.body_text;
```
Comment (chiedoclaude) [inline mobile/src/screens/approvals/EmailThreadCard.tsx:30]: **[Low · Code Quality]** `trimmed` is computed for the emptiness check but the *original* un-trimmed `m.body_text` is returned. Same pattern on line 32 with `sn`/`m.snippet`.

If preserving surrounding whitespace is intentional (reasonable for email bodies), a short comment clarifies that:
```ts
// trim only for empty-check; return original to preserve email whitespace
if (trimmed.length > 0) return m.body_text;
```
Otherwise return `trimmed` and `sn` directly.

File: mobile/src/screens/approvals/__tests__/EmailThreadCard.test.tsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    await act(async () => {
+      renderer = create(
+        createElement(EmailThreadCard, { thread: { subject: "", messages: [] } }),
+      );
+    });
+    expect(hasTestId(renderer, "email-thread-empty")).toBe(true);
+  });
+
+  it("shows truncation affordance when truncated is true", async () => {
+    const thread: EmailThread = {
```
Comment (chiedoclaude) [inline mobile/src/screens/approvals/__tests__/EmailThreadCard.test.tsx:82]: **[Low · Code Quality]** The empty-state test passes `{ subject: "", messages: [] }` but does not test `thread={null}`. Both branches reach the same empty-state UI, but the `null` path exercises the optional-chaining guards in `hasThread` and the `useMemo` computation. Worth adding:
```ts
it("shows empty fallback when thread is null", async () => {
  await act(async () => {
    renderer = create(createElement(EmailThreadCard, { thread: null }));
  });
  expect(hasTestId(renderer, "email-thread-empty")).toBe(true);
});
```

File: mobile/src/screens/approvals/__tests__/ApprovalDetailScreen.test.tsx [hunk 4]
```diff
@@ -166,6 +166,46 @@ describe("ApprovalDetailScreen", () => {
...
+    });
+    await act(async () => {
+      renderer = renderDetail(approval);
+    });
+    const json = JSON.stringify(renderer.toJSON());
+    expect(json).toContain("Email thread");
+    expect(json).toContain("Re: Topic");
+    expect(json).toContain("Thread body");
+    expect(json).not.toContain('"label":"email_thread"');
+  });
```
Comment (chiedoclaude) [inline mobile/src/screens/approvals/__tests__/ApprovalDetailScreen.test.tsx:207]: **[Low · Code Quality]** The email-thread integration test covers `google.send_email_reply` only. `microsoft.send_email_reply` is a separate action type in `EMAIL_REPLY_ACTIONS` — a quick mirror test (action type swapped, same assertions) would catch a future accidental removal from the set.

```ts
it("shows email thread card for microsoft.send_email_reply", async () => {
  const approval = makeApproval({
    action: { type: "microsoft.send_email_reply", version: "1", parameters: {} },
    context: { description: "Reply", risk_level: "low",
      details: { email_thread: { subject: "FW: Topic", messages: [] } } },
  });
  await act(async () => { renderer = renderDetail(approval); });
  expect(JSON.stringify(renderer.toJSON())).toContain("Email thread");
});
```

**Comment by chiedoclaude** [comment]
## Code Review Complete

**PR:** #992 — Mobile: email thread preview on approval for email reply actions (#975 phase 3)
**Author:** @chiedocursor
**Rounds completed:** 2/6
**Reason for stopping:** PR was merged during Round 3 wait

### Outstanding Issues
The following findings were unresolved at the time of merge:

- **[Medium]** `EmailThreadCard.tsx:74` — `AttachmentChips` key collision: `${filename}-${size_bytes}` can be non-unique for duplicate attachments. Fix: include array index in key.
- **[Low]** `EmailThreadCard.tsx:30` — `messageBodyText` computes `trimmed`/`sn` but returns un-trimmed originals; variable naming is misleading.
- **[Low]** `EmailThreadCard.test.tsx` — Missing test for `thread={null}`.
- **[Low]** `ApprovalDetailScreen.test.tsx` — Missing `microsoft.send_email_reply` integration test (only Google action is covered).

### Follow-up Issue
https://github.com/supersuit-tech/permission-slip/issues/1000

### Review History
| Round | Inline Comments | Notes |
|-------|----------------|-------|
| 1 | 4 | Initial review — 0 critical, 0 high, 1 medium, 3 low |
| 2 | 0 | Merge commit from main only; no code changes, no new findings |
| 3 | — | PR merged before round completed |

### Overall Assessment
This PR is a solid, well-scoped implementation of the email thread preview feature. The defensive parsing in `parseEmailThreadDetails` is particularly good — it gracefully handles partial or malformed API payloads without throwing. Test coverage spans three layers (utility, card, screen integration) and the accessibility attributes are thorough. The only concern worth a follow-up is the `AttachmentChips` key uniqueness, which is a minor correctness issue that React will warn about in development if duplicate filenames appear. The three low-severity items (variable naming, missing null-thread test, missing Microsoft integration test) are cosmetic and don't affect production behavior.

Overall: clean feature addition, good to have shipped.

---
*Review performed by Claude Code `/review` skill*

**Comment by chiedoclaude** [comment]
Opened a follow-up issue for non-blocking recommendations from this review: https://github.com/supersuit-tech/permission-slip/issues/1000
